src/widget/wcoverartmenu.cpp

```diff
@@ -82,8 +85,28 @@ void WCoverArtMenu::slotChange() {
     coverInfo.source = CoverInfo::USER_SELECTED;
     coverInfo.coverLocation = selectedCoverPath;
     coverInfo.setImage(image);
-    qDebug() << "WCoverArtMenu::slotChange emit" << coverInfo;
-    emit coverInfoSelected(coverInfo);
+
+    QString coverArtCopyFilePath =
+            fileInfo.absoluteFilePath().left(
+                    fileInfo.absoluteFilePath().lastIndexOf('.') + 1) +
+            selectedCoverExtension;
+
+    if (selectedCoverExtension == "gif") {
```

Comment:
I have just realized that this discovers an underlying issue. Instead of reading the cover art file bit by bit, it is uncompressed into an QImage and when writing it is encoded again. In case of a lossy format like jpg this process will put additional noise to the picture, which should not happen. Instead the copy should be done with the compressed data without the encoding decoding round trip.

Reply:
So you may use a QByteArray to keep the compressed data in memory. 
This can be filled with either the file content or the downloaded data from the web. Maybe a tiny wrapper class is handy that holds the data and the file format. I just had a look to the QT classes but there seems to be nothing suitable already there.

Reply:
This is no longer required ..